### PR TITLE
Add an `otel-collector` module

### PR DESCRIPTION
**Pull request by mattmoor**
This produces the image and grants the service account as which it will run permission to publish metrics.

### CODE REVIEW

File: otel-collector/cmd/otel-collector/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	configPath := os.Getenv("KO_DATA_PATH") + "/config.yaml"
+
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
```
Comment (imjasonh) [inline otel-collector/cmd/otel-collector/main.go:17]: Fail on any error?

File: otel-collector/main.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+resource "google_project_iam_member" "metrics-writer" {
+  project = var.project_id
+  role    = "roles/monitoring.metricWriter"
+  member  = "serviceAccount:${var.service-account}"
+}
+
+resource "ko_build" "otel-image" {
+  base_image  = "cgr.dev/chainguard/opentelemetry-collector-contrib:latest"
```
Comment (imjasonh) [inline otel-collector/main.tf:15]: Make this a var?

**Review by imjasonh — APPROVED** [review]

File: otel-collector/main.tf (hunk 2 above)
Comment (mattmoor) [reply otel-collector/main.tf:15]: done

File: otel-collector/cmd/otel-collector/main.go (hunk 1 above)
Comment (mattmoor) [reply otel-collector/cmd/otel-collector/main.go]: ha, copied from @tcnghia I changed it.

File: otel-collector/cmd/otel-collector/main.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	configPath := os.Getenv("KO_DATA_PATH") + "/config.yaml"
+
+	if _, err := os.Stat(configPath); err != nil {
+		log.Fatalf("error checking config.yaml: $v", err)
```
Comment (tcnghia) [inline otel-collector/cmd/otel-collector/main.go:18]:
```suggestion
		log.Fatalf("error checking config.yaml: %v", err)
```

**Pull request merged**